### ISSUE TITLE
PieMenu crashes FreeCAD at startup when FreeCAD is build with QT6. 

### ISSUE THREAD

**Issue by APEbbers** [Grubuntu/PieMenu#98]
When I use PieMenu with a FreeCAD version that is build with QT6 and Python 3.12, FreeCAD crashes at atartup.
The log shows:
![image](https://github.com/user-attachments/assets/bf8eb59a-90f7-436e-8282-fb5e1eae7b89)
When I use a FreeCAd build with QT5, there is no problem.

I'm using Windows 11 and the develop version of FreeCAD 1.0.

Kind regards,

Paul Ebbers

**Comment by Grubuntu** [comment]
Hi

Where did you find your Qt6 version? I'm not even sure if FreeCAD currently supports Qt6?
I don't have a Qt6 version so I can't test it at the moment.

**Comment by APEbbers** [comment]
Hi, I believe the nigthly builds have QT6. or you can compile your own FreeCAD with the latest commits.
I tried to look for the cause meself, but I couldn't find a clear cause.

Just a heads up:tThe new FreeCAD 1.0 wil have QT6 compatibility. 

kind regards Paul

**Comment by hasecilu** [comment]
Hi @Syres916, hope to not bother you, do you think you could help us with this in the future, it's a problem that has been occurring for some months already.

I made compilation with Qt6 with debug mode on and got the trace below using `gdb`, does it give any clue? It mentions a file at the end, anyway I don't know much about Qt.

I'm Using Arch Linux, latest libraries/packages, compiling from `HEAD`. Any other info needed?

<details>
  <summary><b>Click here to show/hide flags</b></summary>

cmake -S "$source_dir" -B "$build_dir" \
	-D BUILD_FLAT_MESH=ON \
	-D OCCT_CMAKE_FALLBACK=1 \
	-D FREECAD_QT_VERSION=6 \
	-D CMAKE_BUILD_TYPE=Release \
	-D CMAKE_C_FLAGS="$CFLAGS -ffat-lto-objects -fPIC -w" \
	-D CMAKE_CXX_FLAGS="$CXXFLAGS -ffat-lto-objects -fPIC -w" \
	-D FREECAD_USE_EXTERNAL_PIVY=ON \
	-D FREECAD_USE_QT_FILEDIALOG=OFF \
	-D FREECAD_BUILD_DOXYGEN=ON \
	-D CMAKE_BUILD_TYPE=Debug \
	-G Ninja
</details>

```gdb
$ gdb /opt/FreeCAD_1_0_Qt6/bin/FreeCAD
GNU gdb (GDB) 15.2
Copyright (C) 2024 Free Software Foundation, Inc.
License GPLv3+: GNU GPL version 3 or later <http://gnu.org/licenses/gpl.html>
This is free software: you are free to change and redistribute it.
There is NO WARRANTY, to the extent permitted by law.
Type "show copying" and "show warranty" for details.
This GDB was configured as "x86_64-pc-linux-gnu".
Type "show configuration" for configuration details.
For bug reporting instructions, please see:
<https://www.gnu.org/software/gdb/bugs/>.
Find the GDB manual and other documentation resources online at:
    <http://www.gnu.org/software/gdb/documentation/>.

For help, type "help".
Type "apropos word" to search for commands related to "word"...
Reading symbols from /opt/FreeCAD_1_0_Qt6/bin/FreeCAD...
(gdb) run
Starting program: /opt/FreeCAD_1_0_Qt6/bin/FreeCAD

This GDB supports auto-downloading debuginfo from the following URLs:
  <https://debuginfod.archlinux.org>
Enable debuginfod for this session? (y or [n]) y
Debuginfod has been enabled.
To make this setting permanent, add 'set debuginfod enabled on' to .gdbinit.
Downloading separate debug info for /lib64/ld-linux-x86-64.so.2
Downloading 13.47 M separate debug info for /usr/lib/libhdf5.so.310
Downloading 92.82 K separate debug info for /usr/lib/libsz.so.2
Downloading 194.53 K separate debug info for /usr/lib/libz.so.1
Downloading 25.68 M separate debug info for /usr/lib/libmpi.so.40
Downloading 16.40 M separate debug info for /usr/lib/libxerces-c-3.2.so
Downloading 2.52 M separate debug info for /usr/lib/libm.so.6
Downloading 23.62 M separate debug info for /usr/lib/libpython3.12.so.1.0
Downloading 1.17 M separate debug info for /usr/lib/libfmt.so.11
Downloading 2.08 M separate debug info for /usr/lib/libQt6Xml.so.6
Downloading 45.91 M separate debug info for /usr/lib/libCoin.so.80
Downloading 1.89 M separate debug info for /usr/lib/libboost_filesystem.so.1.86.0
Downloading 3.60 M separate debug info for /usr/lib/libboost_program_options.so.1.86.0
Downloading 2.66 M separate debug info for /usr/lib/libboost_regex.so.1.86.0
Downloading 7.54 K separate debug info for /usr/lib/libboost_system.so.1.86.0
Downloading 1.33 M separate debug info for /usr/lib/libboost_thread.so.1.86.0
Downloading 8.81 K separate debug info for /usr/lib/libboost_date_time.so.1.86.0
Downloading 266.10 K separate debug info for /usr/lib/libGL.so.1
Downloading 12.56 M separate debug info for /usr/lib/libQt6PrintSupport.so.6
Downloading 2.20 M separate debug info for /usr/lib/libQt6SvgWidgets.so.6
Downloading 9.89 M separate debug info for /usr/lib/libQt6Svg.so.6
Downloading 44.06 M separate debug info for /usr/lib/libQt6Network.so.6
Downloading 11.05 M separate debug info for /usr/lib/libQt6UiTools.so.6
Downloading 1.74 M separate debug info for /usr/lib/libQt6OpenGLWidgets.so.6
Downloading 134.57 M separate debug info for /usr/lib/libQt6Widgets.so.6
Downloading 27.01 M separate debug info for /usr/lib/libQt6OpenGL.so.6
Downloading 189.35 M separate debug info for /usr/lib/libQt6Gui.so.6
Downloading 70.89 K separate debug info for /usr/lib/libspnav.so.0
Downloading 94.98 M separate debug info for /usr/lib/libQt6Core.so.6
Downloading 9.36 M separate debug info for /usr/lib/libc.so.6
[Thread debugging using libthread_db enabled]
Using host libthread_db library "/usr/lib/libthread_db.so.1".
Downloading 3.46 M separate debug info for /usr/lib/libopen-pal.so.80
Downloading 23.73 M separate debug info for /usr/lib/libfabric.so.1
Downloading 6.91 M separate debug info for /usr/lib/libucp.so.0
Downloading 1.38 M separate debug info for /usr/lib/libucs.so.0
Downloading 583.23 K separate debug info for /usr/lib/libevent_core-2.1.so.7
Downloading 16.59 K separate debug info for /usr/lib/libevent_pthreads-2.1.so.7
Downloading 1.33 M separate debug info for /usr/lib/libhwloc.so.15
Downloading 7.95 M separate debug info for /usr/lib/libpmix.so.2
Downloading 99.38 K separate debug info for /usr/lib/libnsl.so.3
Downloading 4.68 M separate debug info for /usr/lib/libcurl.so.4
Downloading 21.87 M separate debug info for /usr/lib/libicui18n.so.75
Downloading 9.67 M separate debug info for /usr/lib/libicuuc.so.75
Downloading 1.56 K separate debug info for /usr/lib/libicudata.so.75
Downloading 10.90 M separate debug info for /usr/lib/libX11.so.6
Downloading 358.56 K separate debug info for /usr/lib/libexpat.so.1
Downloading 512.62 K separate debug info for /usr/lib/libGLX.so.0
Downloading 64.74 K separate debug info for /usr/lib/libOpenGL.so.0
Downloading 368.53 K separate debug info for /usr/lib/libGLdispatch.so.0
Downloading 1.75 M separate debug info for /usr/lib/libgssapi_krb5.so.2
Downloading 214.33 K separate debug info for /usr/lib/libbrotlidec.so.1
Downloading 5.56 M separate debug info for /usr/lib/libzstd.so.1
Downloading 10.27 K separate debug info for /usr/lib/libproxy.so.1
Downloading 12.93 M separate debug info for /usr/lib/libcrypto.so.3
Downloading 249.66 K separate debug info for /usr/lib/libEGL.so.1
Downloading 1022.15 K separate debug info for /usr/lib/libfontconfig.so.1
Downloading 4.61 M separate debug info for /usr/lib/libglib-2.0.so.0
Downloading 19.11 M separate debug info for /usr/lib/libQt6DBus.so.6
Downloading 718.04 K separate debug info for /usr/lib/libxkbcommon.so.0
Downloading 748.45 K separate debug info for /usr/lib/libpng16.so.16
Downloading 36.04 M separate debug info for /usr/lib/libharfbuzz.so.0
Downloading 221.83 K separate debug info for /usr/lib/libmd4c.so.0
Downloading 3.18 M separate debug info for /usr/lib/libfreetype.so.6
Downloading 3.72 M separate debug info for /usr/lib/libsystemd.so.0
Downloading 383.65 K separate debug info for /usr/lib/libdouble-conversion.so.3
Downloading 10.53 K separate debug info for /usr/lib/libb2.so.1
Downloading 1.43 M separate debug info for /usr/lib/libpcre2-16.so.0
Downloading 310.38 K separate debug info for /usr/lib/libucm.so.0
Downloading 1.01 M separate debug info for /usr/lib/libuct.so.0
Downloading 134.19 K separate debug info for /usr/lib/libnuma.so.1
Downloading 81.23 K separate debug info for /usr/lib/libsframe.so.1
Downloading 1.40 M separate debug info for /usr/lib/libudev.so.1
Downloading 655.02 K separate debug info for /usr/lib/libtirpc.so.3
Downloading 457.13 K separate debug info for /usr/lib/libnghttp3.so.9
Downloading 567.41 K separate debug info for /usr/lib/libnghttp2.so.14
Downloading 108.19 K separate debug info for /usr/lib/libidn2.so.0
Downloading 1.07 M separate debug info for /usr/lib/libssh2.so.1
Downloading 53.44 K separate debug info for /usr/lib/libpsl.so.5
Downloading 3.81 M separate debug info for /usr/lib/libssl.so.3
Downloading 500.05 K separate debug info for /usr/lib/libxcb.so.1
Downloading 2.70 M separate debug info for /usr/lib/libkrb5.so.3
Downloading 685.64 K separate debug info for /usr/lib/libk5crypto.so.3
Downloading 24.65 K separate debug info for /usr/lib/libcom_err.so.2
Downloading 140.23 K separate debug info for /usr/lib/libkrb5support.so.0
Downloading 34.54 K separate debug info for /usr/lib/libkeyutils.so.1
Downloading 192.50 K separate debug info for /usr/lib/libresolv.so.2
Downloading 22.26 K separate debug info for /usr/lib/libbrotlicommon.so.1
Downloading 193.54 K separate debug info for /usr/lib/libproxy/libpxbackend-1.0.so
Downloading 1.73 M separate debug info for /usr/lib/libgobject-2.0.so.0
Downloading 1.53 M separate debug info for /usr/lib/libpcre2-8.so.0
Downloading 1019.40 K separate debug info for /usr/lib/libdbus-1.so.3
Downloading 1.34 M separate debug info for /usr/lib/libgraphite2.so.3
Downloading 156.17 K separate debug info for /usr/lib/libbz2.so.1.0
Downloading 97.45 K separate debug info for /usr/lib/libcap.so.2
Downloading 1.47 M separate debug info for /usr/lib/libunistring.so.5
Downloading 29.26 K separate debug info for /usr/lib/libXau.so.6
Downloading 52.94 K separate debug info for /usr/lib/libXdmcp.so.6
Downloading 10.47 M separate debug info for /usr/lib/libgio-2.0.so.0
Downloading 1.31 M separate debug info for /usr/lib/libduktape.so.207
Downloading 120.63 K separate debug info for /usr/lib/libffi.so.8
Downloading 263.89 K separate debug info for /usr/lib/libgmodule-2.0.so.0
Downloading 1.03 M separate debug info for /usr/lib/libmount.so.1
Downloading 1.00 M separate debug info for /usr/lib/libblkid.so.1
FreeCAD 1.1.0, Libs: 1.1.0devR36460 +2644 (Git)
© Juergen Riegel, Werner Mayer, Yorik van Havre and others 2001-2024
FreeCAD is free and open-source software licensed under the terms of LGPL2+ license.
FreeCAD wouldn't be possible without FreeCAD community.

                        #########################++++++++++
                      #########################++++++++++++
                    #########################++++++++++++++
                  #########################++++++++++++++++
                 ########                         +++++++++
                 ########                         +++++++--
                 ########                         +++++----
                 ########                         +++------
                 ########         ------------------------:
                 ########         ----------------------:
                 ########         --------------------:
                 ########         ------------------:
                 ########                 -----------:.
                 ########                 -------------:.
                 ########                 ---------------:
                 ########         -----------------------:
                 ########         ----------------------:
                 #######+         ---------------:
                 #####+++         --------------:
                 ###+++++         --------------:
                 #+++++++         ---------------:
                 ++++++++++++++++-----:-:--------:
                 ++++++++++++++------:    :----:
                 ++++++++++++-------:
                 ++++++++++--------:
   ________                              ______    ______   _______
  /        |                            /      \  /      \ /       \
      /______    ______    ______  /     |/     |      |
    |__  /      \  /      \  /      \   |   /   |__  |  |    |
       |/     |/     |/     |  |             |  |    |
    /   |   /        |       |  |   __      |  |    |
    |     |          /     /   \__/  |  |    |  |__  |
    |     |              |        |      /   |    |      /
   /     /           /     /     /   /    /     /

Downloading 17.88 K separate debug info for /usr/lib/python3.12/lib-dynload/_opcode.cpython-312-x86_64-linux-gnu.so
Downloading 578.23 K separate debug info for /usr/lib/python3.12/lib-dynload/_datetime.cpython-312-x86_64-linux-gnu.so
NOTICE: Addon "/home/uli/.local/share/FreeCAD/Mod/ThreadProfile" disabled by presence of ADDON_DISABLED stopfile
NOTICE: Addon "/home/uli/.local/share/FreeCAD/Mod/FreeCAD-Beginner-Assistant" disabled by presence of ADDON_DISABLED stopfile
NOTICE: Addon "/home/uli/.local/share/FreeCAD/Mod/FreeCAD_WeldFeature" disabled by presence of ADDON_DISABLED stopfile
NOTICE: Addon "/home/uli/.local/share/FreeCAD/Mod/FreeCAD-Reinforcement" disabled by presence of ADDON_DISABLED stopfile
Downloading 293.39 K separate debug info for /usr/lib/python3.12/lib-dynload/math.cpython-312-x86_64-linux-gnu.so
NOTICE: Addon "/home/uli/.local/share/FreeCAD/Mod/FreeCAD_Assembly4" disabled by presence of ADDON_DISABLED stopfile
NOTICE: Addon "/home/uli/.local/share/FreeCAD/Mod/Assembly3" disabled by presence of ADDON_DISABLED stopfile
NOTICE: Addon "/home/uli/.local/share/FreeCAD/Mod/SteelColumn" disabled by presence of ADDON_DISABLED stopfile
NOTICE: Addon "/home/uli/.local/share/FreeCAD/Mod/A2plus" disabled by presence of ADDON_DISABLED stopfile
Interlocking laser cut workbench loaded
NOTICE: Addon "/home/uli/.local/share/FreeCAD/Mod/FreeCAD-Ribbon.backup1730501382.2333288" disabled by presence of ADDON_DISABLED stopfile
OSHAutoDoc workbench (v0.2.3) loaded
Downloading 154.59 K separate debug info for /usr/lib/python3.12/lib-dynload/binascii.cpython-312-x86_64-linux-gnu.so
Downloading 115.99 K separate debug info for /usr/lib/python3.12/lib-dynload/zlib.cpython-312-x86_64-linux-gnu.so
Downloading 71.27 K separate debug info for /usr/lib/python3.12/lib-dynload/_bz2.cpython-312-x86_64-linux-gnu.so
Downloading 128.47 K separate debug info for /usr/lib/python3.12/lib-dynload/_lzma.cpython-312-x86_64-linux-gnu.so
Downloading 874.19 K separate debug info for /usr/lib/liblzma.so.5
Downloading 156.91 K separate debug info for /usr/lib/python3.12/lib-dynload/_struct.cpython-312-x86_64-linux-gnu.so
Downloading 163.75 K separate debug info for /usr/lib/python3.12/lib-dynload/pyexpat.cpython-312-x86_64-linux-gnu.so
Downloading 55.88 K separate debug info for /usr/lib/python3.12/lib-dynload/_bisect.cpython-312-x86_64-linux-gnu.so
Downloading 40.07 K separate debug info for /usr/lib/python3.12/lib-dynload/_random.cpython-312-x86_64-linux-gnu.so
Downloading 1.12 M separate debug info for /usr/lib/python3.12/lib-dynload/_sha2.cpython-312-x86_64-linux-gnu.so
Downloading 219.77 K separate debug info for /usr/lib/python3.12/lib-dynload/_socket.cpython-312-x86_64-linux-gnu.so
Downloading 77.00 K separate debug info for /usr/lib/python3.12/lib-dynload/select.cpython-312-x86_64-linux-gnu.so
Downloading 184.14 K separate debug info for /usr/lib/python3.12/lib-dynload/array.cpython-312-x86_64-linux-gnu.so
Downloading 33.97 K separate debug info for /usr/lib/python3.12/lib-dynload/fcntl.cpython-312-x86_64-linux-gnu.so
Downloading 146.45 K separate debug info for /usr/lib/python3.12/lib-dynload/_posixsubprocess.cpython-312-x86_64-linux-gnu.so
Downloading 143.26 K separate debug info for /usr/lib/python3.12/lib-dynload/_hashlib.cpython-312-x86_64-linux-gnu.so
Downloading 185.09 K separate debug info for /usr/lib/python3.12/lib-dynload/_blake2.cpython-312-x86_64-linux-gnu.so
Downloading 301.64 K separate debug info for /usr/lib/python3.12/lib-dynload/_ssl.cpython-312-x86_64-linux-gnu.so
Downloading 37.83 K separate debug info for /usr/lib/python3.12/lib-dynload/grp.cpython-312-x86_64-linux-gnu.so
Downloading 81.97 K separate debug info for /usr/lib/python3.12/lib-dynload/_heapq.cpython-312-x86_64-linux-gnu.so
Downloading 42.54 K separate debug info for /usr/lib/python3.12/lib-dynload/_queue.cpython-312-x86_64-linux-gnu.so
Downloading 680.56 K separate debug info for /usr/lib/libQt6Concurrent.so.6
Downloading 4.88 M separate debug info for /usr/lib/libyaml-cpp.so.0.8
Downloading 440.80 K separate debug info for /usr/lib/qt6/plugins/platforms/libqwayland-generic.so
Downloading 22.39 M separate debug info for /usr/lib/qt6/plugins/platforms/../../../libQt6WaylandClient.so.6
Downloading 141.44 K separate debug info for /usr/lib/qt6/plugins/platforms/../../../libwayland-client.so.0
Downloading 46.42 K separate debug info for /usr/lib/qt6/plugins/platforms/../../../libwayland-cursor.so.0
[New Thread 0x7fffe44006c0 (LWP 587530)]
Downloading 2.40 M separate debug info for /usr/lib/qt6/plugins/wayland-shell-integration/libxdg-shell.so
Downloading 1.72 M separate debug info for /usr/lib/qt6/plugins/platformthemes/libqt6ct.so
Downloading 915.74 K separate debug info for /usr/lib/libqt6ct-common.so.0.9
Downloading 2.60 M separate debug info for /usr/lib/qt6/plugins/platformthemes/libqxdgdesktopportal.so
[New Thread 0x7fffe3a006c0 (LWP 587647)]
[New Thread 0x7fffe30006c0 (LWP 587648)]
Downloading 742.63 K separate debug info for /usr/lib/qt6/plugins/platforminputcontexts/libcomposeplatforminputcontextplugin.so
Downloading 448.82 K separate debug info for /usr/lib/qt6/plugins/wayland-graphics-integration-client/libqt-plugin-wayland-egl.so
Downloading 1.49 M separate debug info for /usr/lib/qt6/plugins/wayland-graphics-integration-client/../../../libQt6WaylandEglClientHwIntegration.so.6
Downloading 7.23 K separate debug info for /usr/lib/qt6/plugins/wayland-graphics-integration-client/../../../libwayland-egl.so.1
Downloading 903.25 K separate debug info for /usr/lib/libEGL_mesa.so.0
Downloading 37.92 M separate debug info for /usr/lib/libgallium-24.2.6-arch1.1.so
Downloading 102.84 K separate debug info for /usr/lib/libgbm.so.1
Downloading 97.47 K separate debug info for /usr/lib/libglapi.so.0
Downloading 34.61 K separate debug info for /usr/lib/libX11-xcb.so.1
Downloading 38.16 K separate debug info for /usr/lib/libxcb-dri2.so.0
Downloading 159.27 K separate debug info for /usr/lib/libxcb-randr.so.0
Downloading 72.90 K separate debug info for /usr/lib/libxcb-xfixes.so.0
Downloading 24.12 K separate debug info for /usr/lib/libxcb-shm.so.0
Downloading 243.39 K separate debug info for /usr/lib/libdrm.so.2
Downloading 206.45 K separate debug info for /usr/lib/libwayland-server.so.0
Downloading 39.01 K separate debug info for /usr/lib/libxcb-dri3.so.0
Downloading 25.20 K separate debug info for /usr/lib/libxcb-present.so.0
Downloading 57.62 K separate debug info for /usr/lib/libxcb-sync.so.1
Downloading 10.57 K separate debug info for /usr/lib/libxshmfence.so.1
Downloading 386.18 M separate debug info for /usr/lib/libLLVM.so.18.1
Downloading 131.58 K separate debug info for /usr/lib/libsensors.so.5
Downloading 109.29 K separate debug info for /usr/lib/libdrm_radeon.so.1
Downloading 917.51 K separate debug info for /usr/lib/libelf.so.1
Downloading 139.80 K separate debug info for /usr/lib/libdrm_amdgpu.so.1
Downloading 332.65 K separate debug info for /usr/lib/libdrm_intel.so.1
Downloading 607.88 K separate debug info for /usr/lib/libedit.so.0
Downloading 1.70 M separate debug info for /usr/lib/libncursesw.so.6
Downloading 3.64 M separate debug info for /usr/lib/libxml2.so.2
Downloading 114.30 K separate debug info for /usr/lib/libpciaccess.so.0
[New Thread 0x7fffe26006c0 (LWP 588273)]
[New Thread 0x7fffe1c006c0 (LWP 588274)]
[New Thread 0x7fffe12006c0 (LWP 588275)]
[New Thread 0x7fffd96006c0 (LWP 588276)]
[New Thread 0x7fffd8c006c0 (LWP 588277)]
[New Thread 0x7fffc7c006c0 (LWP 588278)]
[New Thread 0x7fffc72006c0 (LWP 588279)]
[New Thread 0x7fffc68006c0 (LWP 588280)]
[New Thread 0x7fffc5e006c0 (LWP 588281)]
[New Thread 0x7fffc54006c0 (LWP 588282)]
[New Thread 0x7fffc4a006c0 (LWP 588283)]
[New Thread 0x7fffbfe006c0 (LWP 588284)]
Downloading 1.05 M separate debug info for /usr/lib/qt6/plugins/styles/libqt6ct-style.so
Downloading 1.09 M separate debug info for /usr/lib/qt6/plugins/imageformats/libqjpeg.so
Downloading 2.02 M separate debug info for /usr/lib/qt6/plugins/imageformats/../../../libjpeg.so.8
Downloading 1.11 M separate debug info for /usr/lib/qt6/plugins/imageformats/libqsvg.so
Downloading 941.62 K separate debug info for /usr/lib/qt6/plugins/imageformats/libqico.so
Downloading 1.12 M separate debug info for /usr/lib/qt6/plugins/imageformats/libqpdf.so
Downloading 25.43 M separate debug info for /usr/lib/qt6/plugins/imageformats/../../../libQt6Pdf.so.6
Downloading 1.28 M separate debug info for /usr/lib/qt6/plugins/imageformats/../../../libopenjp2.so.7
Downloading 916.75 K separate debug info for /usr/lib/qt6/plugins/imageformats/libqgif.so
Downloading 1.64 M separate debug info for /usr/lib/qt6/plugins/iconengines/libqsvgicon.so
[New Thread 0x7fffbee006c0 (LWP 588651)]
[New Thread 0x7fffbe4006c0 (LWP 588652)]
[New Thread 0x7fffbda006c0 (LWP 588653)]
[New Thread 0x7fffbd0006c0 (LWP 588654)]
[New Thread 0x7fffb3e006c0 (LWP 588655)]
[New Thread 0x7fffb34006c0 (LWP 588656)]
[New Thread 0x7fffb2a006c0 (LWP 588657)]
[New Thread 0x7fffb20006c0 (LWP 588658)]
[New Thread 0x7fffb16006c0 (LWP 588659)]
[New Thread 0x7fffb0c006c0 (LWP 588660)]
[Thread 0x7fffb0c006c0 (LWP 588660) exited]
[Thread 0x7fffb16006c0 (LWP 588659) exited]
[Thread 0x7fffb20006c0 (LWP 588658) exited]
[Thread 0x7fffb2a006c0 (LWP 588657) exited]
[Thread 0x7fffb34006c0 (LWP 588656) exited]
[Thread 0x7fffb3e006c0 (LWP 588655) exited]
Downloading 236.63 K separate debug info for /usr/lib/python3.12/lib-dynload/_json.cpython-312-x86_64-linux-gnu.so
Downloading 310.59 K separate debug info for /usr/lib/python3.12/lib-dynload/_elementtree.cpython-312-x86_64-linux-gnu.so
Downloading 719.66 K separate debug info for /usr/lib/python3.12/lib-dynload/_ctypes.cpython-312-x86_64-linux-gnu.so
NOTICE: Addon "/home/uli/.local/share/FreeCAD/Mod/ThreadProfile" disabled by presence of ADDON_DISABLED stopfile
Downloading 21.91 M separate debug info for /usr/lib/python3.12/site-packages/pivy/_coin.so

Thread 1 "FreeCAD" received signal SIGSEGV, Segmentation fault.
Downloading 37.73 K source file /usr/src/debug/qt6-base/qtbase/src/widgets/kernel/qboxlayout.cpp
QBoxLayoutPrivate::setupGeom (this=this@entry=0x555558dc12d0) at /usr/src/debug/qt6-base/qtbase/src/widgets/kernel/qboxlayout.cpp:253
253             QSize max = box->item->maximumSize();
(gdb)

```

**Comment by Grubuntu** [comment]
I had a lot of trouble compiling FreeCAD with Qt6 for Windows, but finally I got a version that works more or less well for testing.

I finally figured out what was wrong with PieMenu:
 in Qt5 the tab numbering can start at 1 whereas with Qt6 it crashes if it doesn't start at 0... so I patched the current version of PieMenu 1.9.1 to avoid a crash when FreeCAD Qt6 starts. 

https://github.com/Grubuntu/PieMenu/commit/326942dc8dd30401916355343f68223c5cdc7c98

However, there are now a lot of other problems with Qt6:



```
self.menu.setWindowFlags(QtCore.Qt.FramelessWindowHint | Qt.NoDropShadowWindowHint | QtCore.Qt.WA_MacAlwaysShowToolWindow)
During initialization the error "unsupported operand type(s) for |: 'WindowType' and 'WidgetAttribute'
```


```
PieMenuInstance.add_commands(actions, context, text)
22:09:11                                   ^^^^^^^
22:09:11  UnboundLocalError: cannot access local variable 'actions' where it is not associated with a value
```
```

22:09:58      modifier_text = self.get_modifier_text(event.modifiers())
22:09:58                      ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
22:09:58    File "C:\Users\Victus\AppData\Roaming\FreeCAD\Mod\PieMenu\.\InitGui.py", line 246, in get_modifier_text
22:09:58      for modifier in modifier_names if modifiers & modifier])
22:09:58                                        ~~~~~~~~~~^~~~~~~~~~
22:09:58  TypeError: unsupported operand type(s) for &: 'KeyboardModifier' and 'Key'
```

 I'll be working on them over the next few weeks.

Thanks

**Comment by hasecilu** [comment]
Cool, another error means progress. Certainly still not totally fixed but no longer crashes, thanks.
